src/test/java/net/starlark/java/eval/ScriptTest.java

```diff
@@ -108,6 +108,21 @@ public Struct mutablestruct(Dict<String, Object> kwargs) throws EvalException {
     return new MutableStruct(kwargs);
   }
 
+  @StarlarkMethod(
+      name = "unsafe_shallow_freeze",
+      documented = false,
+      parameters = {
+        @Param(name = "x"),
+      }
+  )
+  public void unsafeShallowFreeze(Object object) throws EvalException {
+    if (object instanceof Mutability.Freezable) {
+      ((Mutability.Freezable) object).unsafeShallowFreeze();
```

Comment:
unsafeShallowFreeze is a hack I'd like to get rid of, so I'm not wild about adding new uses of it, even if only to our internal tests. This test could be expressed in Java code.

Reply:
I can rewrite the test in java code, but how about adding a `freeze_for_test` function to tests, which will freeze the whole thread?

Reply:
Fair enough; the Go implementation uses this in its tests. Let's call it freeze(x).